src/main/java/com/greedy/mokkoji/api/scheduler/service/RecruitmentNotificationScheduler.java

```diff
@@ -32,6 +34,37 @@ public void sendDailyRecruitmentNotifications() {
         recruitments.addAll(recruitmentRepository.findAllByRecruitEndToday(today));
         recruitments.addAll(recruitmentRepository.findAllByRecruitEndInThreeDays(threeDaysLater));
 
+        // 1. Club을 키로, Recruitment를 값으로 하는 Map을 생성 (중복 제거 및 최신 선택)
+        Map<Club, Recruitment> latestRecruitmentsByClub = new HashMap<>();
+
+        for (Recruitment recruitment : recruitments) {
+            Club club = recruitment.getClub();
+
+            // 이미 해당 Club에 대한 Recruitment가 Map에 있다면
+            if (latestRecruitmentsByClub.containsKey(club)) {
+                Recruitment existingRecruitment = latestRecruitmentsByClub.get(club);
+
+                // 현재 것이 더 최신이면 Map의 값을 업데이트
+                if (recruitment.getCreatedAt().isAfter(existingRecruitment.getCreatedAt())) {
+                    latestRecruitmentsByClub.put(club, recruitment);
+                }
+            } else {
+                // Map에 없으면 새로운 항목으로 추가
+                latestRecruitmentsByClub.put(club, recruitment);
+            }
+        }
+
+        // 2. Map의 값(가장 최신 Recruitment들)만 추출하여 List로 다시 구성
+        List<Recruitment> uniqueAndLatestRecruitments = new ArrayList<>(latestRecruitmentsByClub.values());
+
+        // 3. 원래의 recruitments 변수에 새 List를 대입하여 forEach문의 입력으로 사용
+        recruitments = uniqueAndLatestRecruitments;
+
+        recruitments.forEach(recruitment -> {
+            Club club = recruitment.getClub();
+            notificationService.sendNotification(club, recruitment);
+        });
+
```

Comment:
```suggestion
         List<Recruitment> uniqueAndLatestRecruitments = recruitments.stream()
            .collect(Collectors.toMap(
                Recruitment::getClub,
                recruitment -> recruitment,
                (recruitment1, recruitment2) -> recruitment1.getCreatedAt().isAfter(recruitment2.getCreatedAt()) ? recruitment1 : recruitment2
            ))
            .values()
            .stream()
            .toList();

        uniqueAndLatestRecruitments.forEach(recruitment -> {
            Club club = recruitment.getClub();
            notificationService.sendNotification(club, recruitment);
        });

```
Stream을 사용해서 간결하게 작성해보는 것은 어떨까요?

Reply:
저도 코드가 더 간결해지면 좋겠다는 의견에 동의합니다!

stream 방식도 깔끔하긴 하지만, 혹시 stream말고 이렇게 if문 하나를 사용하는 방식은 어떠세요??
가독성을 위해서라면 아래처럼 if문 하나만 남긴 방식이 더 직관적이지 않을 까 싶어서요!
ex)
``` java
Map<Club, Recruitment> latestRecruitmentsByClub = new HashMap<>();

for (Recruitment recruitment : recruitments) {
    Club club = recruitment.getClub();
    Recruitment existing = latestRecruitmentsByClub.get(club);

    if (existing == null || recruitment.getCreatedAt().isAfter(existing.getCreatedAt())) {
        latestRecruitmentsByClub.put(club, recruitment);
    }
}

recruitments = new ArrayList<>(latestRecruitmentsByClub.values());

recruitments.forEach(r -> notificationService.sendNotification(r.getClub(), r));

```

Reply:
물론, 대량 데이터를 다루는 데에는 stream이 적합하긴 하지만,
저희 비교 로릭은 단순한 편이라 stream에서 발생하는 박싱·언박싱이나
중간 연산 생성 등이 오히려 현재 로직 대비 다소 무거운 처리 방식이 아닐까 생각이 듭니다.

Reply:
저는 가독성을 고려했을 때 두 분 의견이 모두 적절하다고 생각합니다!

다만, 이전부터 저희가 이러한 부분을 stream으로 처리해왔기 때문에 해당 부분에서 일관성을 유지하는 것에 더 의미를 둔다면 stream으로 처리하면 좋을 수도 있다는 생각이 듭니다!
하지만 이 부분은 생각하기 나름이라 두 분의 의견이 더 궁금합니다 : )

Reply:
좋아요! stream 보다 가독성이 좋은 것 같네요. 해당 방식으로 수정해주시면 감사하겠습니다 :)

Reply:
``` java
BinaryOperator<Recruitment> selectLatest =
        (r1, r2) -> r1.getCreatedAt().isAfter(r2.getCreatedAt()) ? r1 : r2;

List<Recruitment> latestRecruitments = recruitments.stream()
        .collect(Collectors.toMap(
                Recruitment::getClub,
                r -> r,
                selectLatest
        ))
        .values()
        .stream()
        .toList();

latestRecruitments.forEach(r ->
        notificationService.sendNotification(r.getClub(), r)
);
```

Reply:
음... 일관성은 유지해야하는 데 가독성이 고민이라면..
이렇게 최신 모집공고를 찾는 부분을 함수로 빼는 게 어떠세요?
오히려 이게 더 복잡하려나용...

Reply:
Stream의 장점이 다양한 메서드를 통해 로직의 전체 흐름을 빠르게 이해할 수 있다는 점이라고 생각합니다!
함수로 분리하게 된다면 구현 내용을 보기 위해 한 번 내려갔다가 올라와야 해서 stream의 장점을 오히려 못 살린다고 판단했어용
좀 복잡한 로직일 경우 분리하는 게 좋겠지만, 해당 함수는 빠르게 이해 가능한 로직이라 그냥 두는 것도 좋다고 생각합니다!!

하지만, 너무 가독성을 해친다고 생각되신다면 함수명을 좀 더 구체적으로 짓고 분리하는 건 어떠신가요?

Reply:
음... 함수로 분리해야할 정도로 가독성을 해친다고는 생각이 들지 않네요.
그럼 금서님이 적어주신 코드대로 stream을 택하돼, 저기에 상시모집 공고를 제외하는 로직을 추가하는 거 어떠신가요?
현재 상시모집 공고를 제외하는 부분이 없어서요!

Reply:
``` java
List<Recruitment> uniqueAndLatestRecruitments = recruitments.stream()
    .filter(recruitment -> !recruitment.isAlwaysOpen()) // <- 상시모집 제외
    .collect(Collectors.toMap(
        Recruitment::getClub,
        recruitment -> recruitment,
        (recruitment1, recruitment2) -> 
            recruitment1.getCreatedAt().isAfter(recruitment2.getCreatedAt())
                ? recruitment1
                : recruitment2
    ))
    .values()
    .stream()
    .toList();

uniqueAndLatestRecruitments.forEach(recruitment -> {
    Club club = recruitment.getClub();
    notificationService.sendNotification(club, recruitment);
});
```
이런식으로요!!

Reply:
네 좋습니다! 그렇게 진행해주시면 될 것 같아요